march_hardware/include/march_hardware/MarchRobot.h

```diff
@@ -0,0 +1,70 @@
+// Copyright 2019 Project March.
+
+#ifndef MARCH4CPP__MARCH4_H
+#define MARCH4CPP__MARCH4_H
+
+#include <string>
+#include <vector>
+#include <stdint.h>
+
+#include <march_hardware/Joint.h>
+
+#include <march_hardware/EtherCAT/EthercatMaster.h>
+
+namespace march4cpp
+{
+class MarchRobot
+{
+private:
+  std::unique_ptr<EthercatMaster> ethercatMaster;
+
+public:
+  ::std::vector<Joint> jointList;
+
+  MarchRobot(::std::vector<Joint> jointList, ::std::string ifName, int ecatCycleTime);
+
+  MarchRobot();
+
+  void startEtherCAT();
+
+  void stopEtherCAT();
+
+  int getMaxSlaveIndex();
+
+  bool hasValidSlaves();
+
+  bool isEthercatOperational();
+
+  Joint getJoint(::std::string jointName);
+
+  /** @brief Override comparison operator */
+  friend bool operator==(const MarchRobot& lhs, const MarchRobot& rhs)
+  {
+    if (lhs.jointList.size() != rhs.jointList.size())
+    {
+      return false;
+    }
+    for (unsigned int i = 0; i < lhs.jointList.size(); i++)
+    {
+      const march4cpp::Joint lhsJoint = lhs.jointList.at(i);
+      const march4cpp::Joint rhsJoint = rhs.jointList.at(i);
+      if (lhsJoint != rhsJoint)
+      {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  /** @brief Override stream operator for clean printing */
+  friend ::std::ostream& operator<<(std::ostream& os, const MarchRobot& marchRobot)
+  {
+    for (unsigned int i = 0; i < marchRobot.jointList.size(); i++)
+    {
+      os << marchRobot.jointList.at(i) << "\n";
+    }
+    return os;
```

Comment:
Here we 'overload' the == (comperator) operator between 2 classes. So this function gets called whenever you run `robot1 == robot2`. It will compare them with the implementation of this function where robot1 will become the lhs and robot2 the rhs. 'lhs' and 'rhs' are convention for such binary operator overload functions.

More info:
https://stackoverflow.com/questions/16843323/c-object-equality
https://en.cppreference.com/w/cpp/language/operators